modules/templates/mailer.go

```diff
@@ -64,7 +64,14 @@ func Mailer(ctx context.Context) (*texttmpl.Template, *template.Template) {
 	bodyTemplates := template.New("")
 
 	subjectTemplates.Funcs(mailSubjectTextFuncMap())
-	bodyTemplates.Funcs(NewFuncMap())
+	// To do the best to avoid serious breaking, add some functions back for body templates.
+	// Keep in mind that some behaviors have changed, for worse case, double-escaping.
+	// Custom template users should migrate to the new template system ASAP.
+	bodyTemplateFuncMap := NewFuncMap()
+	bodyTemplateFuncMap["Safe"] = SafeHTML
```

Comment:
Maybe add a future version to remove them. How about 1.23?

Reply:
Maybe we can have a warning log when user use those functions.

Reply:
I just doubt whether there are enough end users really reading warning logs .....

And you see there are a lot of "warning" in code:

```
log.Error("Deprecated setting `[markup.*]` `DISABLE_SANITIZER` present. This fallback will be removed in v1.18.0")
```

Since 1.18. Have you ever seen any of them really been removed?

Reply:
I have removed some of them.

Reply:
> I have removed some of them.

Which one? I take a quick look on app.example.ini:  https://github.com/go-gitea/gitea/commits/main/custom/conf/app.example.ini?author=lunny

Only "cache.ENABLED" is removed because the cache is always enabled, it isn't a deprecation, nor a breaking.

Reply:
In d2d1314d1871462af504302329976da1b43b06d5, I removed all fallbacks, just fatal. So it is a clear "breaking" change now.